test/unit/mem/view/src/ViewSubViewTest.cpp

```diff
@@ -45,132 +45,208 @@
+    template<
+        typename TAcc,
+        typename TElem>
+    auto viewSubViewTestOffset()
+    -> void
+    {
+        using Dev = alpaka::dev::Dev<TAcc>;
+        using Pltf = alpaka::pltf::Pltf<Dev>;
 
-    //-----------------------------------------------------------------------------
-    alpaka::test::mem::view::viewTestMutable<
-        TAcc>(
-            queue,
-            view);
+        using Dim = alpaka::dim::Dim<TAcc>;
+        using Idx = alpaka::idx::Idx<TAcc>;
+        using View = alpaka::mem::view::ViewSubView<Dev, TElem, Dim, Idx>;
 
-    //-----------------------------------------------------------------------------
-    // alpaka::mem::view::traits::GetPitchBytes
-    // The pitch of the view has to be identical to the pitch of the underlying buffer in all dimensions.
-    {
-        auto const pitchBuf(alpaka::mem::view::getPitchBytesVec(buf));
-        auto const pitchView(alpaka::mem::view::getPitchBytesVec(view));
+        Dev const dev(alpaka::pltf::getDevByIdx<Pltf>(0u));
 
-        for(Idx i = Dim::value; i > static_cast<Idx>(0u); --i)
-        {
-            BOOST_REQUIRE_EQUAL(
-                pitchBuf[i-static_cast<Idx>(1u)],
-                pitchView[i-static_cast<Idx>(1u)]);
-        }
-    }
+        auto const extentBuf(alpaka::vec::createVecFromIndexedFnWorkaround<Dim, Idx, CreateExtentBufVal>(Idx()));
+        auto buf(alpaka::mem::buf::alloc<TElem, Idx>(dev, extentBuf));
 
-    //-----------------------------------------------------------------------------
-    // alpaka::mem::view::traits::GetPtrNative
-    // The native pointer has to be exactly the value we calculate here.
-    {
-        auto viewPtrNative(
-            reinterpret_cast<std::uint8_t *>(
-                alpaka::mem::view::getPtrNative(buf)));
-        auto const pitchBuf(alpaka::mem::view::getPitchBytesVec(buf));
-        for(Idx i = Dim::value; i > static_cast<Idx>(0u); --i)
-        {
-            auto const pitch = (i < static_cast<Idx>(Dim::value)) ? pitchBuf[i] : static_cast<Idx>(sizeof(Elem));
-            viewPtrNative += offsetView[i - static_cast<Idx>(1u)] * pitch;
-        }
-        BOOST_REQUIRE_EQUAL(
-            reinterpret_cast<Elem *>(viewPtrNative),
-            alpaka::mem::view::getPtrNative(view));
+        auto const extentView(alpaka::vec::createVecFromIndexedFnWorkaround<Dim, Idx, CreateExtentViewVal>(Idx()));
+        auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(sizeof(Idx)));
+        View view(buf, extentView, offsetView);
+
+        alpaka::test::mem::view::viewSubViewTest<TAcc>(view, buf, dev, extentView, offsetView);
     }
 }
+}
+}
+}
 #if BOOST_COMP_GNUC
     #pragma GCC diagnostic pop
 #endif
 
+BOOST_AUTO_TEST_SUITE(memView)
+
+//-----------------------------------------------------------------------------
+BOOST_AUTO_TEST_CASE_TEMPLATE(
+    viewSubViewNoOffsetTest,
+    TAcc,
+    alpaka::test::acc::TestAccs)
+{
+    alpaka::test::mem::view::viewSubViewTestNoOffset<TAcc, float>();
+}
```

Comment:
The diff looks like much has changed, but in the end only this single test case is new and the code from the other test case has been extracted into a function that can be reused.